cmake/modules/FindGnuLd.cmake

```
@@ -0,0 +1,59 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2022, Nordic Semiconductor ASA
+# Copyright (c) 2023, Intel Corporation
+
+# FindGnuLd module for locating GNU ld (linker from binutils).
+#
+# The module defines the following variables:
+#
+# 'GNULD'
+# Path to GNU ld linker
+# Set to 'GNULD-NOTFOUND' if ld was not found.
+#
+# 'GnuLd_FOUND', 'GNULD_FOUND'
+# True if GNU ld was found.
+#
+# 'GNULD_VERSION_STRING'
+# The version of GNU ld.
+
+if(DEFINED TOOLCHAIN_HOME)
+  # When Toolchain home is defined, then we are cross-compiling, so only look
+  # for linker in that path, else we are using host tools.
+  set(LD_SEARCH_PATH PATHS ${TOOLCHAIN_HOME} NO_DEFAULT_PATH)
+endif()
+
+find_program(GNULD ${CROSS_COMPILE}ld.bfd ${LD_SEARCH_PATH})
+if(NOT GNULD)
+  find_program(GNULD ${CROSS_COMPILE}ld ${LD_SEARCH_PATH})
+endif()
+
+if(GNULD)
+  # Parse the 'ld.bfd --version' output to find the installed version.
+  execute_process(
+    COMMAND
+    ${GNULD} --version
+    OUTPUT_VARIABLE gnuld_version_output
+    ERROR_VARIABLE  gnuld_error_output
+    RESULT_VARIABLE gnuld_status
+    )
+
+  set(GNULD_VERSION_STRING)
+  if(${gnuld_status} EQUAL 0)
+    # Extract GNU ld version. Different distros have their
+    # own version scheme so we need to account for that.
+    # Examples:
+    # - "GNU ld (GNU Binutils for Ubuntu) 2.34"
+    # - "GNU ld (Zephyr SDK 0.15.2) 2.38"
+    # - "GNU ld (Gentoo 2.39 p5) 2.39.0"
+    string(REGEX MATCH
+           "GNU ld \\(.+\\) ([0-9]+[.][0-9]+[.]?[0-9]*).*"
```

Maybe a bit stricter?
```suggestion
           "GNU ld \\(.+\\) ([0-9]+[.][0-9]+[.]?[0-9]*)$"
```
What happens when there's no match? Did you locally break the regex and see how that's reported?

---

The version string would simply be empty. Cmake will not print the version, and any version comparisons will be false.